packages/backend/src/open_payments/payment_pointer/service.ts

```diff
@@ -102,6 +119,29 @@ async function createPaymentPointer(
   }
 }
 
+async function updatePaymentPointer(
+  deps: ServiceDependencies,
+  { id, status, publicName }: UpdateOptions
+): Promise<PaymentPointer | PaymentPointerError> {
+  try {
+    const update: UpdateInput = { publicName }
+    if (status) {
+      update.deactivatesAt =
+        status === PaymentPointerStatus.Inactive ? new Date() : null
+    }
+
+    return await PaymentPointer.query(deps.knex)
+      .patchAndFetchById(id, update)
+      .withGraphFetched('asset')
+      .throwIfNotFound()
+  } catch (err) {
+    if (err instanceof NotFoundError) {
+      return PaymentPointerError.UnknownAsset
```

Comment:
We need to define a new error for that case. `UnknownAsset` is for when you are trying to create a payment pointer with an asset (e.g. USD, scale 2), and that asset does not exist.

Reply:
added a more appropriate error